### ISSUE TITLE
In-text disambiguation of first authors using initials sometimes uses non-first authors

### ISSUE THREAD

**Issue by xwhatsit** [plk/biblatex-apa#98]
This occurs with both biblatex-apa v8.0 (APA 6 style) and v9.6 (APA 7 style).

The in-text disambiguation of first authors (using initials) works as expected, aside from the fact it also looks at second (and third etc.) sometimes.

Example:
```bibtex
@article{bloggsAndSmith2001,
	title = {Title 1},
	author = {Bloggs, Joe and Smith, John},
	date = {2001},
}

@article{smith2002,
	title = {Title 2},
	author = {Smith, Adam},
	date = {2002},
}

@article{smith2003,
	title = {Title 3},
	author = {Smith, Adam},
	date = {2003},
}
```

```latex
\documentclass{article}
\usepackage[style=apa]{biblatex}
\addbibresource{bib.bib}

\begin{document}

\cite{bloggsAndSmith2001}

\cite{smith2002}

\cite{smith2003}

\printbibliography
\end{document}

```

This produces:

> Bloggs and Smith, 2001
> A. Smith, 2002
> A. Smith, 2003
>
> **References**
> Bloggs, J., & Smith, J. (2001). Title 1.
> Smith, A. (2002). Title 2.
> Smith, A. (2003). Title 3.

As far as I can understand, A. Smith should not need the initials added to the in-text citation.

Interestingly, removing one of the references (any of them, for all other permutations) produces expected results, e.g.:

```latex
\documentclass{article}
\usepackage[style=apa]{biblatex}
\addbibresource{bib.bib}

\begin{document}

\cite{bloggsAndSmith2001}

\cite{smith2002}

\printbibliography
\end{document}
```
> Bloggs and Smith, 2001
> Smith, 2002
>
> **References**
> Bloggs, J., & Smith, J. (2001). Title 1.
> Smith, A. (2002). Title 2.

or:

```latex
\documentclass{article}
\usepackage[style=apa]{biblatex}
\addbibresource{bib.bib}

\begin{document}

\cite{smith2002}

\cite{smith2003}

\printbibliography
\end{document}

```

> Smith, 2002
> Smith, 2003
>
> **References**
> Smith, A. (2002). Title 2.
> Smith, A. (2003). Title 3.

**Comment by plk** [comment]
This was actually a fairly obscure `biber` bug which has been corrected version 2.15 DEV on Sourceforge.

**Comment by xwhatsit** [comment]
The development biber version indeed fixes the issue for this 600+ item reference list. Thank-you so much!
